### PR TITLE
Modal: Refactor to Handle onClose Properly

### PR DESCRIPTION
**Pull request by westwood846**
This MR refactors the Modal component to remove the internal state, which before has been completely controlled by the `isVisible` prop, which made it redundant.

This MR also refactors the way `onClose` is called. Previously, the `componentWillUnmount` was supposed to unregister the escape key event handler, so that the event would not trigger while the modal was closed. This did not work because the modal's visibility is toggled through the `display` CSS prop, so the component would never actually unmount.

### CODE REVIEW

File: src/Display/Modal/Modal.tsx [hunk 1]
```diff
@@ -13,9 +13,10 @@ import {
   ModalFooter,
 } from '../../Style/Display/ModalStyle';
 
-class Modal extends React.Component<Props, State> {
+class Modal extends React.Component<Props> {
   modalContentAreaRef: React.RefObject<HTMLDivElement>;
   mouseDownTarget: HTMLElement;
+  escEvent: (e: KeyboardEvent) => any;
 
   state = {
```
Comment (tingtingdisco) [inline src/Display/Modal/Modal.tsx]: state can be removed

File: src/Display/Modal/Modal.tsx [hunk 2]
```diff
@@ -26,34 +27,35 @@ class Modal extends React.Component<Props, State> {
...
-    document.addEventListener('keydown', escEvent(onClose), false);
-  }
+  componentDidUpdate(prevProps: Props) {
+    const { isVisible } = this.props;
 
-  static getDerivedStateFromProps(nextProps: Props, prevState: State) {
-    if (nextProps.isVisible && !prevState.isOpen) {
+    // closed -> open
+    if (!prevProps.isVisible && isVisible) {
+      this.modalContentAreaRef.current.focus();
```
Comment (tingtingdisco) [inline src/Display/Modal/Modal.tsx]: can we remove the `modalContentAreaRef`?

File: src/Display/Modal/Modal.tsx (hunk 1 above)
Comment (westwood846) [reply src/Display/Modal/Modal.tsx]: Removed it!

File: src/Display/Modal/Modal.tsx (hunk 2 above)
Comment (westwood846) [reply src/Display/Modal/Modal.tsx]: I did not touch that logic in this PR. No idea what that's about in the first place haha.

Comment (tingtingdisco) [reply src/Display/Modal/Modal.tsx]: yeah, there seems to be some dead code and unnecessary code, like the `escEvent` which is essentially an alias of `onClose`, maybe can be removed as well.

File: src/Display/Modal/Modal.tsx [hunk 3]
```diff
@@ -13,47 +13,49 @@ import {
...
   }
 
   componentWillUnmount() {
+    this.handleClose();
+  }
+
+  handleOpen() {
     const { onClose } = this.props;
-    document.removeEventListener('keydown', escEvent(onClose), false);
+    this.escEvent = escEvent(onClose);
```
Comment (tingtingdisco) [inline src/Display/Modal/Modal.tsx]: can do `this.escEvent = escEvent(onClose);` in constructor?

File: src/Display/Modal/Modal.tsx [hunk 4]
```diff
@@ -13,47 +13,49 @@ import {
...
+      this.modalContentAreaRef.current.focus();
       document.body.style.overflow = 'hidden';
-      return { isOpen: true };
+      this.handleOpen();
     }
 
-    if (!nextProps.isVisible && prevState.isOpen) {
+    // open -> closed
+    if (prevProps.isVisible && !isVisible) {
       document.body.removeAttribute('style');
```
Comment (tingtingdisco) [inline src/Display/Modal/Modal.tsx]: this line is in `handleClose` already, can be removed

File: src/Display/Modal/Modal.tsx [hunk 5]
```diff
@@ -13,47 +13,49 @@ import {
...
-  }
+  componentDidUpdate(prevProps: Props) {
+    const { isVisible } = this.props;
 
-  static getDerivedStateFromProps(nextProps: Props, prevState: State) {
-    if (nextProps.isVisible && !prevState.isOpen) {
+    // closed -> open
+    if (!prevProps.isVisible && isVisible) {
+      this.modalContentAreaRef.current.focus();
       document.body.style.overflow = 'hidden';
```
Comment (tingtingdisco) [inline src/Display/Modal/Modal.tsx]: can move `document.body.style.overflow = 'hidden';` to `handleOpen`

**Pull request closed without merging**